ShareExtension/ShareExtension/ViewModels/ShareViewModel.swift

```diff
@@ -12,8 +12,8 @@ class ShareViewModel: ObservableObject {
     @Published var uploadStatus = UploadStatus(success: false)
     @Published var image: UIImage?
 
-    func saveDataToDocuments(_ data: Data?, imageFileName: String = Strings.sharedImageName) {
-        guard let imageFileURL = getDocumentsDirectory()?.appendingPathComponent(imageFileName) else { return }
+    func saveDataToDocuments(_ data: Data?) {
+        let imageFileURL = URL.documentsDirectory.appendingPathComponent("myImage_\(Int.random(in: 1...10000)).jpg")
```

Comment:
Create a function in a file that extends the string and name it like generate random image name and replace it here.
i.e. 
extension String {
func generateRandomImageName() -> String {
  let prefix = "myImage"
  let randomNumber = Int.random(in: 1...10000)
  let suffix = ".jpg"
  
  return prefix + randomNumber + suffix
}
}